sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2378,10 +2375,6 @@ class Analyzer(override val catalogManager: CatalogManager)
      */
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUpWithPruning(
       _.containsPattern(PLAN_EXPRESSION), ruleId) {
-      // In case of HAVING (a filter after an aggregate) we use both the aggregate and
-      // its child for resolution.
-      case f @ Filter(_, a: Aggregate) if f.childrenResolved =>
```

Comment:
This is not related to this PR but I just thought about the following case:

```sql
-- t1: [c1, c2]: (0,1),(1,2) , t: [a, b]: (0,1),(1,2)
select sum(c2) as c1 from t1 group by c1 having (select sum(b) from t where c1 = a) > 0
```
Because ResolveSubquery comes before ResolveAggregateFunction, this `c1` in `c1 = a` will be resolved as `sum(c2)` instead of `t1.c1`, because UnresolvedHaving is also a UnaryNode which will be resolved under the UnaryNode case below. So the result here can be incorrect.

Here we need to exclude UnresolvedHaving from this rule.

Reply:
Good catch! Since it's an existing bug, let's fix it separately with a test.